app/routes/wallet/sd_jws.py

```diff
@@ -23,16 +23,114 @@
-
-See https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-07.html for the SD-JWT / SD-JWS spec.
-""",
 )
 async def sign_sd_jws(
     body: SDJWSCreateRequest,
     auth: AcaPyAuth = Depends(acapy_auth_from_header),
 ) -> SDJWSCreateResponse:
+    """
+    Sign Selective Disclosure for JWS (SD-JWS)
+    ---
+
+    This endpoint allows the user to sign a Selective Disclosure for JWS (SD-JWS).
+    The endpoint returns the SD-JWS that can be used to selectively disclose
+    attributes to a verifier.
+
+    When populating the the body of the request, the user must populate either the `did`
+    or the `verification_method` field.
+
+    If an issuer sings a JWS with a `did:sov` DID, the DID should be public.
+
+    The difference between the `did` and `verification_method` fields is
+    that if the `did` field is used, the Aries agent will make an educated guess
+    about which verkey associated with the DID to use to sign the jwt.
+
+    While with the `verification_method` field, the user is explicitly
+    specifying which verkey to use to sign the jwt, i.e. the DID with the key to use.
+
+    The `header` field is optional and can be used to specify the header of the JWS.
+    The `typ`, `alg`, and `kid` fields, in the header, are automatically populated by the Aries agent.
+
+    The `non_sd_list` field is a list of non-selective disclosure attributes.
+    These are attributes that are not included in the selective disclosure i.e.
+    they are always disclosed.
+    If an attribute is either a list or a dictionary, the attribute name should be
+    in the `non_sd_list`.
+    Then the attribute will be disclosed in the SD-JWS but not the values in the list or
+    attributes value pairs in the dictionary.
+
+    The values in a list can be added to the `non_sd_list` by adding the attribute name and
+    the range of the list to the `non_sd_list` in the format:
+        `"<attribute_name>[<start>:<end>]"`
+    where `<start>` is the start of the range and `<end>` is the end of the range
+    (where <end> is exclusive).
+
+    The values in a dictionary can be added to the `non_sd_list` by adding the dictionary name
+    dot the attribute name to the `non_sd_list` in the format:
+        `"<dictionary_name>.<attribute_name>"`
+    where `<dictionary_name>` is the name of the dictionary and `<attribute_name>` is the
+    attribute name in the dictionary.
+
+    The endpoint will return the signed SD-JWS with disclosures needed to reveal the
+    attributes in the SD-JWS.
+        `<Issuer-signed JWS>~<Disclosure 1>~<Disclosure 2>~...~<Disclosure N>~`
+
+    Its up to the holder to identify which disclosure matches with which attributes in the SD-JWS.
+    As the holder will need to pass on the SD-JWS with the correct disclosures to the verifier when
+    requested.
+
+    See https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-07.html
+    for the SD-JWT / SD-JWS spec.
```

Comment:
This is good, but I think it's intimidatingly long. Maybe it can stay this long, but just send it through some GPT proof reading with the goal of making it more concise where possible. I guess we don't want to trim away important details, but if some of this can be referenced to a doc elsewhere, would be more readable